tests/functions/slicer_test.py

```diff
@@ -72,3 +72,9 @@ def test_with_mismatching_data_arrays_raises():
     b.coords['xx'] *= 1.1
     with pytest.raises(sc.DatasetError):
         _ = Slicer({'a': a, 'b': b}, keep=['xx'])
+
+
+def test_raises_ValueError_when_given_binned_data():
```

Comment:
slicer is not importing `check_not_binned`, it's using `preprocess` where the check is performed. Do we need this test or do we have enough with `plot` and `scatter3d`?

Reply:
Do we have a test, testing that it calls `proprocess`? Even if we did, I think that would be a bad idea. We should not test implementation details but expected *behaviors* (observable by a user).

Reply:
If  we are testing behaviours (which I agree is the correct thing to do), then you should also add tests for `inspector` and `superplot`, to cover everything?

Reply:
Hmm, as you will be adding more cases, how can we handle this? Can we somehow create a suite of common tests that we run on all "plot" functions?

Reply:
> as you will be adding more cases, how can we handle this

yeah, that's why I thought you intentionally left them out because they were using `preprocess` ;-)

Reply:
> Can we somehow create a suite of common tests that we run on all "plot" functions?

make a parametrized test where the parameters are the different plotting functions?

Reply:
The problem is that they may need different types of inputs and parameters?

Reply:
Make the parameters a dict or tuple paring function with args?
But then, it's maybe just as unmaintainable as writing a test for each function?